iceoryx_hoofs/include/iceoryx_hoofs/internal/units/duration.inl

```diff
@@ -461,53 +466,123 @@ inline constexpr Duration Duration::operator*(const T& rhs) const noexcept
 {
     static_assert(std::is_arithmetic<T>::value, "non arithmetic types are not supported for multiplication");
 
-    if (rhs <= static_cast<T>(0) || *this == Duration::zero())
+    if ((rhs <= static_cast<T>(0)) || (*this == Duration::zero()))
     {
         return Duration::zero();
     }
 
     return multiplyWith<T>(rhs);
 }
 
-
+// AXIVION Next Construct AutosarC++19_03-A8.4.7 : Each argument is larger than two words
 template <typename T>
 inline constexpr Duration operator*(const T& lhs, const Duration& rhs) noexcept
 {
     return rhs * lhs;
 }
 
+
+/// @brief Equal to operator
+/// @param[in] lhs is the left hand side of the comparison
+/// @param[in] rhs is the right hand side of the comparison
+/// @return true if duration equal to rhs
+// AXIVION Next Line AutosarC++19_03-A8.4.7 : Each argument is larger than two words
```

Comment:
Please move the doxygen documentation to the `.hpp`.

Reply:
@mossmaurice These are not member functions. Are you suggesting to put the prototype for these free functions in the `duration.hpp` header ?

Reply:
Correct, that's what I meant. In iceoryx -for templates- declaration and doxygen go into `.hpp` and definition into `.inl`.